src/HttpMethodsClient.php

```diff
@@ -79,12 +92,55 @@ public function options($uri, array $headers = [], $body = null): ResponseInterf
 
     public function send(string $method, $uri, array $headers = [], $body = null): ResponseInterface
     {
-        return $this->sendRequest($this->requestFactory->createRequest(
-            $method,
-            $uri,
-            $headers,
-            $body
-        ));
+        if (!is_string($uri) && !$uri instanceof UriInterface) {
+            throw new \TypeError(
+                sprintf('%s::send(): Argument #2 ($uri) must be of type string|%s, %s given', self::class, UriInterface::class, get_debug_type($uri))
+            );
+        }
+
+        if (!is_string($body) && !$body instanceof StreamInterface && null !== $body) {
+            throw new \TypeError(
+                sprintf('%s::send(): Argument #4 ($body) must be of type string|%s|null, %s given', self::class, StreamInterface::class, get_debug_type($body))
+            );
+        }
+
+        return $this->sendRequest(
+            self::createRequest($method, $uri, $headers, $body)
+        );
+    }
+
+    /**
+     * @param string|UriInterface         $uri
+     * @param string|StreamInterface|null $body
+     */
+    private function createRequest(string $method, $uri, array $headers = [], $body = null): RequestInterface
+    {
+        if ($this->requestFactory instanceof RequestFactory) {
+            return $this->requestFactory->createRequest(
+                $method,
+                $uri,
+                $headers,
+                $body
+            );
+        }
+
+        if (is_string($body) && null === $this->streamFactory) {
+            throw new \RuntimeException('Cannot create request: A stream factory is required to create a request with a string body.');
```

Comment:
the strict BC way would be to ignore the body, right? though that very much sounds like a bug, so the "BC break" is that you now realize that your code is not working as expected.

but we should also ignore `''` then, to avoid unnecessarily breaking systems that happened to send an empty string.

Reply:
Yeh, we could ignore the empty string too.

Reply:
Done!